hwmgr-plugins/controller/hardwareplugin_controller.go

```diff
@@ -99,9 +104,111 @@ func (r *HardwarePluginReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	return
 }
 
+// setupOAuthClientConfig constructs an OAuth client configuration from the HardwarePlugin CR.
+func (r *HardwarePluginReconciler) setupOAuthClientConfig(ctx context.Context, hwplugin *pluginv1alpha1.HardwarePlugin) (*sharedutils.OAuthClientConfig, error) {
+	var caBundle string
+	if hwplugin.Spec.CaBundleName != nil {
+		cm, err := sharedutils.GetConfigmap(ctx, r.Client, *hwplugin.Spec.CaBundleName, hwplugin.Namespace)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get CA Bundle configmap: %w", err)
+		}
+
+		caBundle, err = sharedutils.GetConfigMapField(cm, sharedutils.CABundleFilename)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get certificate bundle from configmap: %w", err)
+		}
+	}
+
+	config := sharedutils.OAuthClientConfig{
+		TLSConfig: &sharedutils.TLSConfig{CaBundle: []byte(caBundle)},
+	}
+
+	if hwplugin.Spec.AuthClientConfig.TLSConfig != nil && hwplugin.Spec.AuthClientConfig.TLSConfig.SecretName != nil {
+		secretName := *hwplugin.Spec.AuthClientConfig.TLSConfig.SecretName
+		cert, key, err := sharedutils.GetKeyPairFromSecret(ctx, r.Client, secretName, hwplugin.Namespace)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get certificate and key from secret: %w", err)
+		}
+
+		config.TLSConfig.ClientCert = sharedutils.NewStaticKeyPairLoader(cert, key)
```

Comment:
Using a static loader here is ok if this client is only ever going to be used once for validation, but if this is a long-lived client that we'll use to talk to the plugin then we'll need to handle this differently with something that can refresh the token periodically whenever the secret is replaced with an updated certificate.  That's best handled if the secret is mounted to a volume and we can use a dynamic cert loader.   We can discuss what's needed here offline.

Reply:
The client will only be used once for validation - hence static loader.